[BUG][typescript-angular] Configuration template generates code that fails strict type checking

<!--
Please follow the issue template below for bug reports and feature requests.
Also please indicate in the issue title which language/library is concerned. Eg:  [JAVA] Bug generating foo with bar 
-->

##### Description

The code generated by the typescript-angular template results in code that fails the `strictNullChecks` compiler option for Typescript

##### openapi-generator version

5.0.1 - issue added in 5.0.0 via  #6953

##### OpenAPI declaration file content or url
```
openapi: 3.0.0
info:
  version: '1.0'
  title: Example API
paths:
  /securitySchemeA:
    get:
      responses:
        default:
          description: successful operation
components:
  securitySchemes:
    ExampleA:
      type: apiKey
      name: Authorization
      in: header
```

##### Command line used for generation

`openapi-generator generate -g typescript-angular -i openapi.yml`

##### Steps to reproduce

Use the spec file and command provided above to generate a client.
Then run the command `npx tsc configuration.ts --lib esnext --strictNullChecks`.
Ignore the error `Cannot find module '@angular/common/http'`.
Observe two errors stating `Object is possibly 'undefined'`

##### Related issues/PRs

Not that I can see

##### Suggest a fix/enhancement

https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/typescript-angular/configuration.mustache#L67
The code could be changed to use [Optional Chaining (?.)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) to access the keys on `this.apiKeys`, since `apiKeys` is nullable. e.g. `this.apiKeys?.['{{name}}']`
Alternatively, it could simply check for the truthiness of `apiKeys` before accessing it.

--- a/modules/openapi-generator/src/main/resources/typescript-angular/configuration.mustache
+++ b/modules/openapi-generator/src/main/resources/typescript-angular/configuration.mustache
@@ -64,7 +64,11 @@ export class {{configurationClassName}} {
 {{#isApiKey}}
             this.credentials['{{name}}'] = () => {
                 {{! Fallback behaviour may be removed for 5.0 release. See #5062 }}
-                return this.apiKeys?.['{{name}}'] || this.apiKeys?.['{{keyParamName}}'];
+                if (this.apiKeys === null || this.apiKeys === undefined) {
+                    return undefined;
+                } else {
+                    return this.apiKeys['{{name}}'] || this.apiKeys['{{keyParamName}}'];
+                }
             };
 {{/isApiKey}}
 {{#isBasic}}

--- a/modules/openapi-generator/src/main/resources/typescript-angular/configuration.mustache
+++ b/modules/openapi-generator/src/main/resources/typescript-angular/configuration.mustache
@@ -64,7 +64,7 @@ export class {{configurationClassName}} {
 {{#isApiKey}}
             this.credentials['{{name}}'] = () => {
                 {{! Fallback behaviour may be removed for 5.0 release. See #5062 }}
-                return this.apiKeys['{{name}}'] || this.apiKeys['{{keyParamName}}'];
+                return this.apiKeys?.['{{name}}'] || this.apiKeys?.['{{keyParamName}}'];
             };
 {{/isApiKey}}
 {{#isBasic}}

--- a/samples/client/petstore/typescript-angular-v10-provided-in-root/builds/default/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v10-provided-in-root/builds/default/configuration.ts
@@ -61,7 +61,11 @@ export class Configuration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
+                if (this.apiKeys === null || this.apiKeys === undefined) {
+                    return undefined;
+                } else {
+                    return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                }
             };
         }
 

--- a/samples/client/petstore/typescript-angular-v10-provided-in-root/builds/default/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v10-provided-in-root/builds/default/configuration.ts
@@ -61,7 +61,7 @@ export class Configuration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
             };
         }
 

--- a/samples/client/petstore/typescript-angular-v10-provided-in-root/builds/with-npm/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v10-provided-in-root/builds/with-npm/configuration.ts
@@ -61,7 +61,11 @@ export class Configuration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
+                if (this.apiKeys === null || this.apiKeys === undefined) {
+                    return undefined;
+                } else {
+                    return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                }
             };
         }
 

--- a/samples/client/petstore/typescript-angular-v10-provided-in-root/builds/with-npm/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v10-provided-in-root/builds/with-npm/configuration.ts
@@ -61,7 +61,7 @@ export class Configuration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
             };
         }
 

--- a/samples/client/petstore/typescript-angular-v11-provided-in-root/builds/default/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v11-provided-in-root/builds/default/configuration.ts
@@ -61,7 +61,11 @@ export class Configuration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
+                if (this.apiKeys === null || this.apiKeys === undefined) {
+                    return undefined;
+                } else {
+                    return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                }
             };
         }
 

--- a/samples/client/petstore/typescript-angular-v11-provided-in-root/builds/default/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v11-provided-in-root/builds/default/configuration.ts
@@ -61,7 +61,7 @@ export class Configuration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
             };
         }
 

--- a/samples/client/petstore/typescript-angular-v11-provided-in-root/builds/with-npm/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v11-provided-in-root/builds/with-npm/configuration.ts
@@ -61,7 +61,11 @@ export class Configuration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
+                if (this.apiKeys === null || this.apiKeys === undefined) {
+                    return undefined;
+                } else {
+                    return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                }
             };
         }
 

--- a/samples/client/petstore/typescript-angular-v11-provided-in-root/builds/with-npm/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v11-provided-in-root/builds/with-npm/configuration.ts
@@ -61,7 +61,7 @@ export class Configuration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
             };
         }
 

--- a/samples/client/petstore/typescript-angular-v6-not-provided-in-root/builds/default/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v6-not-provided-in-root/builds/default/configuration.ts
@@ -61,7 +61,11 @@ export class Configuration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
+                if (this.apiKeys === null || this.apiKeys === undefined) {
+                    return undefined;
+                } else {
+                    return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                }
             };
         }
 

--- a/samples/client/petstore/typescript-angular-v6-not-provided-in-root/builds/default/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v6-not-provided-in-root/builds/default/configuration.ts
@@ -61,7 +61,7 @@ export class Configuration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
             };
         }
 

--- a/samples/client/petstore/typescript-angular-v6-not-provided-in-root/builds/with-npm/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v6-not-provided-in-root/builds/with-npm/configuration.ts
@@ -61,7 +61,11 @@ export class Configuration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
+                if (this.apiKeys === null || this.apiKeys === undefined) {
+                    return undefined;
+                } else {
+                    return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                }
             };
         }
 

--- a/samples/client/petstore/typescript-angular-v6-not-provided-in-root/builds/with-npm/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v6-not-provided-in-root/builds/with-npm/configuration.ts
@@ -61,7 +61,7 @@ export class Configuration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
             };
         }
 

--- a/samples/client/petstore/typescript-angular-v6-provided-in-root/builds/default/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v6-provided-in-root/builds/default/configuration.ts
@@ -61,7 +61,11 @@ export class Configuration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
+                if (this.apiKeys === null || this.apiKeys === undefined) {
+                    return undefined;
+                } else {
+                    return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                }
             };
         }
 

--- a/samples/client/petstore/typescript-angular-v6-provided-in-root/builds/default/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v6-provided-in-root/builds/default/configuration.ts
@@ -61,7 +61,7 @@ export class Configuration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
             };
         }
 

--- a/samples/client/petstore/typescript-angular-v6-provided-in-root/builds/with-npm/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v6-provided-in-root/builds/with-npm/configuration.ts
@@ -61,7 +61,11 @@ export class Configuration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
+                if (this.apiKeys === null || this.apiKeys === undefined) {
+                    return undefined;
+                } else {
+                    return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                }
             };
         }
 

--- a/samples/client/petstore/typescript-angular-v6-provided-in-root/builds/with-npm/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v6-provided-in-root/builds/with-npm/configuration.ts
@@ -61,7 +61,7 @@ export class Configuration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
             };
         }
 

--- a/samples/client/petstore/typescript-angular-v7-not-provided-in-root/builds/default/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v7-not-provided-in-root/builds/default/configuration.ts
@@ -61,7 +61,11 @@ export class Configuration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
+                if (this.apiKeys === null || this.apiKeys === undefined) {
+                    return undefined;
+                } else {
+                    return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                }
             };
         }
 

--- a/samples/client/petstore/typescript-angular-v7-not-provided-in-root/builds/default/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v7-not-provided-in-root/builds/default/configuration.ts
@@ -61,7 +61,7 @@ export class Configuration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
             };
         }
 

--- a/samples/client/petstore/typescript-angular-v7-not-provided-in-root/builds/with-npm/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v7-not-provided-in-root/builds/with-npm/configuration.ts
@@ -61,7 +61,11 @@ export class Configuration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
+                if (this.apiKeys === null || this.apiKeys === undefined) {
+                    return undefined;
+                } else {
+                    return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                }
             };
         }
 

--- a/samples/client/petstore/typescript-angular-v7-not-provided-in-root/builds/with-npm/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v7-not-provided-in-root/builds/with-npm/configuration.ts
@@ -61,7 +61,7 @@ export class Configuration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
             };
         }
 

--- a/samples/client/petstore/typescript-angular-v7-provided-in-root/builds/default/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v7-provided-in-root/builds/default/configuration.ts
@@ -61,7 +61,11 @@ export class Configuration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
+                if (this.apiKeys === null || this.apiKeys === undefined) {
+                    return undefined;
+                } else {
+                    return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                }
             };
         }
 

--- a/samples/client/petstore/typescript-angular-v7-provided-in-root/builds/default/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v7-provided-in-root/builds/default/configuration.ts
@@ -61,7 +61,7 @@ export class Configuration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
             };
         }
 

--- a/samples/client/petstore/typescript-angular-v7-provided-in-root/builds/with-npm/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v7-provided-in-root/builds/with-npm/configuration.ts
@@ -61,7 +61,11 @@ export class Configuration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
+                if (this.apiKeys === null || this.apiKeys === undefined) {
+                    return undefined;
+                } else {
+                    return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                }
             };
         }
 

--- a/samples/client/petstore/typescript-angular-v7-provided-in-root/builds/with-npm/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v7-provided-in-root/builds/with-npm/configuration.ts
@@ -61,7 +61,7 @@ export class Configuration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
             };
         }
 

--- a/samples/client/petstore/typescript-angular-v8-provided-in-root/builds/single-request-parameter/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v8-provided-in-root/builds/single-request-parameter/configuration.ts
@@ -61,7 +61,11 @@ export class Configuration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
+                if (this.apiKeys === null || this.apiKeys === undefined) {
+                    return undefined;
+                } else {
+                    return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                }
             };
         }
 

--- a/samples/client/petstore/typescript-angular-v8-provided-in-root/builds/single-request-parameter/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v8-provided-in-root/builds/single-request-parameter/configuration.ts
@@ -61,7 +61,7 @@ export class Configuration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
             };
         }
 

--- a/samples/client/petstore/typescript-angular-v8-provided-in-root/builds/with-npm/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v8-provided-in-root/builds/with-npm/configuration.ts
@@ -61,7 +61,11 @@ export class Configuration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
+                if (this.apiKeys === null || this.apiKeys === undefined) {
+                    return undefined;
+                } else {
+                    return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                }
             };
         }
 

--- a/samples/client/petstore/typescript-angular-v8-provided-in-root/builds/with-npm/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v8-provided-in-root/builds/with-npm/configuration.ts
@@ -61,7 +61,7 @@ export class Configuration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
             };
         }
 

--- a/samples/client/petstore/typescript-angular-v8-provided-in-root/builds/with-prefixed-module-name/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v8-provided-in-root/builds/with-prefixed-module-name/configuration.ts
@@ -61,7 +61,11 @@ export class PetStoreConfiguration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
+                if (this.apiKeys === null || this.apiKeys === undefined) {
+                    return undefined;
+                } else {
+                    return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                }
             };
         }
 

--- a/samples/client/petstore/typescript-angular-v8-provided-in-root/builds/with-prefixed-module-name/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v8-provided-in-root/builds/with-prefixed-module-name/configuration.ts
@@ -61,7 +61,7 @@ export class PetStoreConfiguration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
             };
         }
 

--- a/samples/client/petstore/typescript-angular-v9-provided-in-any/builds/default/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v9-provided-in-any/builds/default/configuration.ts
@@ -61,7 +61,11 @@ export class Configuration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
+                if (this.apiKeys === null || this.apiKeys === undefined) {
+                    return undefined;
+                } else {
+                    return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                }
             };
         }
 

--- a/samples/client/petstore/typescript-angular-v9-provided-in-any/builds/default/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v9-provided-in-any/builds/default/configuration.ts
@@ -61,7 +61,7 @@ export class Configuration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
             };
         }
 

--- a/samples/client/petstore/typescript-angular-v9-provided-in-root/builds/default/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v9-provided-in-root/builds/default/configuration.ts
@@ -61,7 +61,11 @@ export class Configuration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
+                if (this.apiKeys === null || this.apiKeys === undefined) {
+                    return undefined;
+                } else {
+                    return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                }
             };
         }
 

--- a/samples/client/petstore/typescript-angular-v9-provided-in-root/builds/default/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v9-provided-in-root/builds/default/configuration.ts
@@ -61,7 +61,7 @@ export class Configuration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
             };
         }
 

--- a/samples/client/petstore/typescript-angular-v9-provided-in-root/builds/with-npm/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v9-provided-in-root/builds/with-npm/configuration.ts
@@ -61,7 +61,11 @@ export class Configuration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
+                if (this.apiKeys === null || this.apiKeys === undefined) {
+                    return undefined;
+                } else {
+                    return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                }
             };
         }
 

--- a/samples/client/petstore/typescript-angular-v9-provided-in-root/builds/with-npm/configuration.ts
+++ b/samples/client/petstore/typescript-angular-v9-provided-in-root/builds/with-npm/configuration.ts
@@ -61,7 +61,7 @@ export class Configuration {
         // init default api_key credential
         if (!this.credentials['api_key']) {
             this.credentials['api_key'] = () => {
-                return this.apiKeys['api_key'] || this.apiKeys['api_key'];
+                return this.apiKeys?.['api_key'] || this.apiKeys?.['api_key'];
             };
         }
 

